### DISCUSSION TITLE
Tests with real console window

### DISCUSSION THREAD

**Discussion by kindermannhubert** [waf/PrettyPrompt#45] [General]
I'm inclined to add a new test project which would be a normal console application that would run a set of tests that cannot be done in standard unit tests. It's limiting because it cannot be run automatically on Github but better to have manually runnable tests than nothing.

I've for example tried to write a unit test for #30 but unsuccessfully. I think it depends on the specific implementation of ```IConsole```.

I'm not even sure that the behavior described in #30 is the same on all platforms (I'm Windows only 🙂).

@waf - do you agree to have such test project?

**Comment by kindermannhubert** [comment]
It occurred to me now that this new app could just record states (possibly different for different platforms) of ```IConsole``` (the real one = implemented by ```System.Console```) for a given input/resizing/scrolling. These states would be deserialized and used in normal unit tests.
This approach would be probably the best although more complex.

**Comment by waf** [comment]
I agree it'd be nice to have some sort of integration test like this. I've definitely run into issues where the underlying System.Console seems to have unexpected behavior.

Either manipulating a real terminal window (Windows-only is fine) or some sort of serialized state approach would be fine. I would personally lean towards the "real terminal" approach over the "serialized state" approach, since if the goal is to simulate weird terminal behavior we may as well go all the way and use real terminal. Whatever approach works best is fine by me, though.

**Comment by kindermannhubert** [comment]
Integration testing with real window and record/replay capabilities implemented in #230.
